### PR TITLE
Add localizing cht initial commit

### PR DESCRIPTION
**Pull request by billwambua**
REF https://github.com/medic/app-services-team/issues/33

### CODE REVIEW

File: content/en/apps/reference/localizing-cht.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+---
+title: "CHT Localization/"
+linkTitle: "cht-localization/"
+weight: 2
+description: >
+  **Localization**: Localizing the CHT Application
+keywords: localization languages translations
+---
+
+Given that CHT apps are used around the world, the Core Framework was designed with localization in mind. The Core Framework itself is available in English, French, Hindi, Nepali, Spanish, Swahili, and Indonesian. In the `app_settings.json` file the default language for the application is set, along with a separate default language for outgoing messages that are sent via SMS.
```
Comment (kennsippell) [inline content/en/apps/reference/localizing-cht.md:10]: This is a great document on tutorials, guides, explanation, and reference. https://documentation.divio.com/

You currently have this listed as reference but the task in https://github.com/medic/app-services-team/issues/33 is to create a localiation tutorial.  The good news is that you seem to have strong elements here for a great tutorial here!  

Can I challenge you to look at existing cht-docs tutorials, check out that guide for writing tutorials, and reform that content and excellent illustration to fit that pattern?

**Review by kennsippell — DISMISSED** [review]

File: content/en/apps/tutorials/localizing-cht.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+
+**Note**: The CHV can also choose the language of their choice when they login for the first time. A popup appears where they can choose their preferred language. 
+### 3. Outgoing Texts
+
+These are the SMS notifications/replies that go to CHVs and Supervisors phones in projects that incorporate SMS workflows.
+
+To change the language of outgoing texts to a particular CHV/Supervisor to Swahili: 
+
+i). First in app settings, when configuring replies, add Swahili(sw) under `locales`. Like so:
```
Comment (derickl) [inline content/en/apps/tutorials/localizing-cht.md:120]: app settings => app_settings.json

File: content/en/apps/tutorials/localizing-cht.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+```
+
+#### 2. App Forms
+To localize an app form to say the Kiswahili language, open the appropriate xlsx of the form and add a `label::sw` column which has the translation for the text. This will work in the `Survey` sheet or the `choices` sheet.
+
+![configuration](app-forms-localization.png)
+
+
+_**App Management - Admin Area**_
+This will still remain in English even after changing the default language.
```
Comment (derickl) [inline content/en/apps/tutorials/localizing-cht.md:109]: This defaults to english and is unaffected by any locale change.

Could you please confirm if this is a bug or is by design?

File: content/en/apps/tutorials/localizing-cht.md [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+e.g to change the `Number of fever cases managed` target to Swahili on the instance, adding the appropriate translation in the `messages-sw.properties` file. i.e
+
+```
+targets.fever_cases_managed.title = Idadi ya kesi za homa kusimamiwa
+```
+
+#### 2. App Forms
+To localize an app form to say the Kiswahili language, open the appropriate xlsx of the form and add a `label::sw` column which has the translation for the text. This will work in the `Survey` sheet or the `choices` sheet.
+
+![configuration](app-forms-localization.png)
```
Comment (derickl) [inline content/en/apps/tutorials/localizing-cht.md:105]: Let's also show the two form renderings from the app

File: content/en/apps/tutorials/localizing-cht.md [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+          ],
+          "event_type": "report_accepted",
+          "recipient": "reporting_unit"
+        },
+``` 
+iii) Change the CHVs language in app management > users > [Choose CHV username e.g chv_1] > Language 
+
+![configuration](change-user-language.png)
+
+### Translating CHT to another language
```
Comment (derickl) [inline content/en/apps/tutorials/localizing-cht.md:167]: Is there a plan to add steps to translate targets?

File: content/en/apps/tutorials/localizing-cht.md [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+
+## Implementation Steps
+
+Create a new file in the 'translations/' folder called messages-sw.properties
+
+1. Localize CHT Elements to Swahili
+_**CHT Instance text**_ - for non-admin users, this is the text that falls under **Messages**, **Tasks**, **Reports**, **People** and **Targets**.
+To localize instance text to Swahili, change the default system language to `Swahili`.
+
+![configuration](change-system-language.png)
```
Comment (derickl) [inline content/en/apps/tutorials/localizing-cht.md:41]: Consider adding step by step screenshots that correspond to the text instructions.

File: content/en/apps/tutorials/localizing-cht.md [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+
+Use the instructions outlined [here](https://docs.communityhealthtoolkit.org/apps/reference/translations/#translations) to  change the instance language.
```
[the remark this answers is not in the document]
Comment (derickl) [reply content/en/apps/tutorials/localizing-cht.md:43]: Tutorials should be prescriptive. As such a user would only need to follow the instructions to achieve the intended outcome. From you are instructions, you've mostly highlighted what to change but not how to effect it. e.g which cht-conf command do you run?

**Review by derickl — COMMENTED** [review]
It's shaping up well @billwambua 

I've left some initial thoughts inline.

Also, consider running hugo locally to test out the docs site from the repo

File: content/en/apps/tutorials/localizing-cht.md (hunk 2 above)
Comment (kennsippell) [reply content/en/apps/tutorials/localizing-cht.md:128]: base_settings.json

File: content/en/apps/tutorials/localizing-cht.md (hunk 3 above)
Comment (billwambua) [reply content/en/apps/tutorials/localizing-cht.md]: This is consistent across versions and is therefore by design.

Comment (derickl) [reply content/en/apps/tutorials/localizing-cht.md]: Could you please open a cht-core issue? I'm assuming if folks want the app in french, then the whole app should be in french (including admin)

File: content/en/apps/tutorials/localizing-cht.md (hunk 5 above)
Comment (billwambua) [reply content/en/apps/tutorials/localizing-cht.md:175]: https://github.com/medic/cht-docs/blob/31a32b05f75020c2686b18dc7a2a8f5f90db5b1b/content/en/apps/tutorials/localizing-cht.md?plain=1#L101

**Review by derickl — DISMISSED** [review]
I've left a few more comments inline.

Overall, it's shaping up well @billwambua . There would be value delving a bit deeper about how one would use the App Management > Display > Languages view to identify which translation keys affect which parts of the app.

File: content/en/apps/tutorials/localizing-cht.md (hunk 4 above)
Comment (billwambua) [reply content/en/apps/tutorials/localizing-cht.md]: It's `label::sw`. Not sure what the question is referring to here since neither `label::id` nor `label::en` is in the doc.

Comment (derickl) [reply content/en/apps/tutorials/localizing-cht.md]: label::id is indonesian. so it's OK. I'd lost that context

File: content/en/apps/tutorials/localizing-cht.md [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+```
+cht --local upload-custom-translations
+```
+The default Swahili translations that come pre-added to CHT can be found [here](https://github.com/medic/cht-core/blob/master/ddocs/medic/_attachments/translations/messages-sw.properties).
+
+After changing the instance language to Swahili, the various elements will behave like this:
+
+_**Messages**_
+
+In Messages, the time counter text and text that shows the user how to navigate the tab changes.
```
[the remark this answers is not in the document]
Comment (billwambua) [reply content/en/apps/tutorials/localizing-cht.md:64]: I have highlighted the changes in the image.

File: content/en/apps/tutorials/localizing-cht.md [hunk 9]
```diff
@@ -0,0 +1,232 @@
...
+
+Create a new file in the 'translations/' folder called `messages-sw.properties`.
+
+After an edit or addition of a translation, upload the current `messages-sw.properties` onto your local environment using the below command.
+
+```
+cht --url=https://medic:password@localhost --upload-custom-translations
+```
+
+### 1. Localize CHT Elements to Swahili
```
Comment (smbuthia) [inline content/en/apps/tutorials/localizing-cht.md:43]:
```suggestion
### 1. Add User Interface Label Translations
```
Make the heading a little more descriptive.

File: content/en/apps/tutorials/localizing-cht.md [hunk 10]
```diff
@@ -0,0 +1,232 @@
...
+
+![configuration](messages-tab-language-en.png)
+
+*Swahili Translation*
+
+![configuration](messages-tab-language-sw.png)
+
+_**Tasks**_
+
+You can localize the task header by adding the appropriate translation in the `messages-sw.properties` file.
```
Comment (smbuthia) [inline content/en/apps/tutorials/localizing-cht.md:78]:
```suggestion
Localize the task header by adding the appropriate translation in the `messages-sw.properties` file.
```

File: content/en/apps/tutorials/localizing-cht.md [hunk 11]
```diff
@@ -0,0 +1,232 @@
...
+
+e.g to translate the below delivery task title to Swahili
```
Comment (smbuthia) [inline content/en/apps/tutorials/localizing-cht.md:80]:
```suggestion
For example, to translate the below delivery task title to Swahili:
```

File: content/en/apps/tutorials/localizing-cht.md [hunk 12]
```diff
@@ -0,0 +1,232 @@
...
+```
+{
+    name: 'anc-home-visit-delivery',
+    icon: 'icon-pregnancy',
+    title: 'task.anc.delivery.title',
+``` 
+
+Add the below code to the `messages-sw.properties` file
```
Comment (smbuthia) [inline content/en/apps/tutorials/localizing-cht.md:88]:
```suggestion
Add the code below to the `messages-sw.properties` file
```

File: content/en/apps/tutorials/localizing-cht.md [hunk 13]
```diff
@@ -0,0 +1,232 @@
...
+![configuration](localize-tasks-en.png)
+
+*Swahili translation*
+![configuration](localize-tasks-sw.png)
+
+_**Reports**_
+
+You can localize the report field names by adding the appropriate translation in the `messages-{language-code}.properties` file.
+
+e.g to change the date of birth field to Swahili, in `messages-sw.properties` file, add this:
```
Comment (smbuthia) [inline content/en/apps/tutorials/localizing-cht.md:104]:
```suggestion
Localize the report field names by adding the appropriate translation in the `messages-{language-code}.properties` file.

For example, to change the date of birth field to Swahili, in `messages-sw.properties` file, add this:
```

File: content/en/apps/tutorials/localizing-cht.md [hunk 14]
```diff
@@ -0,0 +1,232 @@
...
+
+*After
+
+![configuration](people-translation-sw.png)
+
+_**Targets**_
+
+You can localize the names of the targets by adding the appropriate translation in the `messages-{language-code}.properties file.
+
+e.g to add the `Growth Monitoring` target title in Swahili on the instance, add the appropriate translation in the `messages-sw.properties` file. i.e
```
Comment (smbuthia) [inline content/en/apps/tutorials/localizing-cht.md:143]:
```suggestion
For example, to add the `Growth Monitoring` target title in Swahili on the instance, add the appropriate translation in the `messages-sw.properties` file. For instance:
```
Avoid abbreviations whenever you can. Use for example instead of e.g., and for instance instead of i.e.

File: content/en/apps/tutorials/localizing-cht.md [hunk 15]
```diff
@@ -0,0 +1,232 @@
...
+```
+*Before adding the Swahili target label*
+
+![configuration](growth-monitoring-before.png)
+
+*After adding the Swahili target label*
+
+![configuration](growth-monitoring-after.png)
+
+#### 2. App Forms
```
Comment (smbuthia) [inline content/en/apps/tutorials/localizing-cht.md:156]:
```suggestion
### 2. App Forms
```

File: content/en/apps/tutorials/localizing-cht.md [hunk 16]
```diff
@@ -0,0 +1,232 @@
...
+
+![configuration](new-person-form-swahili.png)
+
+
+_**App Management - Admin Area**_
+This will still remain in English even after changing the default language to Swahili.
+
+
+
+**Note**: The CHV can also choose the language of their choice when they login for the first time. A popup appears where they can choose their preferred language.
```
Comment (smbuthia) [inline content/en/apps/tutorials/localizing-cht.md:176]:
```suggestion
{{% alert title="Note" %}}The CHV can also choose the language of their choice when they login for the first time. A popup appears where they can choose their preferred language.{{% /alert %}}
```
Use alert tags for notes and content you want to highlight.

File: content/en/apps/tutorials/localizing-cht.md [hunk 17]
```diff
@@ -0,0 +1,232 @@
...
+
+
+**Note**: The CHV can also choose the language of their choice when they login for the first time. A popup appears where they can choose their preferred language. 
+### 3. Outgoing Texts
+
+These are the SMS notifications/replies that go to CHVs and Supervisors phones in projects that incorporate SMS workflows.
+
+To change the language of outgoing texts to a particular CHV/Supervisor to Swahili: 
+
+i). First in the `app_settings.json`, when configuring replies, add Swahili(sw) under `locales`. Like so:
```
Comment (smbuthia) [inline content/en/apps/tutorials/localizing-cht.md:183]:
```suggestion
First in the `app_settings.json`, when configuring replies, add Swahili(sw) under `locales`:
```

File: content/en/apps/tutorials/localizing-cht.md [hunk 18]
```diff
@@ -0,0 +1,232 @@
...
+      "code": "ne",
+      "name": "Nepali"
+    },
+    {
+      "code": "sw",
+      "name": "Swahili"
+    }
+``` 
+
+ii) Set up the translation for the reply message. For example:
```
Comment (smbuthia) [inline content/en/apps/tutorials/localizing-cht.md:208]:
```suggestion
Set up the translation for the reply message:
```

File: content/en/apps/tutorials/localizing-cht.md [hunk 19]
```diff
@@ -0,0 +1,232 @@
...
+            {
+              "content": "Asante, kuhudhuria kwa {{patient_name}} ({{patient_id}} kumerekodiwa.",
+              "locale": "sw"
+            }
+          ],
+          "event_type": "report_accepted",
+          "recipient": "reporting_unit"
+        },
+``` 
+iii) Change the CHVs language in app management > users > [Choose CHV username e.g chv_1] > Language > Pick Swahili
```
Comment (smbuthia) [inline content/en/apps/tutorials/localizing-cht.md:226]:
```suggestion
Change the CHVs language in app management > users > [Choose CHV username e.g chv_1] > Language > Pick Swahili:
```

**Review by smbuthia — CHANGES_REQUESTED** [review]
Great! I have added some comments and suggestions inline. If you can, also edit the images to a more consistent size to make the doc a bit more aesthetically pleasing for readers.

File: content/en/apps/tutorials/localizing-cht.md [hunk 20]
```diff
@@ -0,0 +1,295 @@
...
+      "code": "sw",
+      "name": "Swahili"
+    }
+``` 
+
+Set up the translation for the reply message:
+```
+"messages": [
+        {
+          "message": [
```
Comment (derickl) [inline content/en/apps/tutorials/localizing-cht.md:275]: indentation is off

**Comment by derickl** [comment]
@smbuthia could you do a review on this?

**Comment by kennsippell** [comment]
@smbuthia Are you the appropriate reviewer for this?

File: content/en/apps/tutorials/localizing-cht.md [hunk 21]
```diff
@@ -0,0 +1,291 @@
+---
+title: "CHT Localization/"
+linkTitle: "Localizing CHT"
```
Comment (smbuthia) [inline content/en/apps/tutorials/localizing-cht.md:3]:
```suggestion
linkTitle: "Localization"
```

**Review by smbuthia — CHANGES_REQUESTED** [review]
@billwambua there's a pending suggestion from before and one I've just added. 

For the images, could you organize them such that the image is next to the related text? Right now it's hard to follow the content against the image. You could add breaks after images too.

**Comment by billwambua** [comment]
@smbuthia I have made the requested changes. Let me know if the image positioning works now.

**Review by smbuthia — APPROVED** [review]
The image alignment is better now. Thanks for making the changes.

**Comment by billwambua** [comment]
@derickl and @kennsippell your approval is needed for merge.

**Comment by derickl** [comment]
Back to you @billwambua
